asmrun/Makefile

```diff
@@ -142,20 +142,26 @@ install:
 $(LINKEDFILES): %.c: ../byterun/%.c
 	$(LN) $< $@
 
-%.d.$(O): %.c
-	$(CC) -c $(DFLAGS) $(CPPFLAGS) $(OUTPUTOBJ)$@ $<
+# The COMPILE_C_FILE function below receives as argument the pattern
+# that corresponds to the name of the generated object file
+define COMPILE_C_FILE
+$(1): %.c
+	$$(CC) -c $$(OC_CFLAGS) $$(OC_CPPFLAGS) $$(OUTPUTOBJ)$$@ $$<
+endef
 
```

Comment:
I find this GNU make macro wizardry very hard to read.  The result of the macro expansion (i.e. 5 implicit rules) would be more readable, not to mention more compact...